File: src/main/java/com/dev/photoCatalog/model/Photoshoot.java
package com.dev.photoCatalog.model;

import jakarta.persistence.*;
import java.sql.Timestamp;

@Entity
public class Photoshoot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer photoshootID;

    @Column(nullable = false)
    private Timestamp date;

    @Column(name = "locationID", nullable = true)
    private Integer locationID;  // Storing locationID as Integer

    // Default constructor
    public Photoshoot() {}

    // Parameterized constructor
    public Photoshoot(Timestamp date, Integer locationID) {
        this.date = date;
        this.locationID = locationID;
    }

    // Getters and Setters
    public Integer getPhotoshootID() {
        return photoshootID;
    }

    public void setPhotoshootID(Integer photoshootID) {
        this.photoshootID = photoshootID;
    }

    public Timestamp getDate() {
        return date;
    }

    public void setDate(Timestamp date) {
        this.date = date;
    }

    public Integer getLocationID() {
        return locationID;
    }

    public void setLocationID(Integer locationID) {
        this.locationID = locationID;
    }
}

File: frontend/src/models/Photoshoot.js
class Photoshoot {
    constructor(photoshootID, date) {
      this.photoshootID = photoshootID; 
      this.date = date; 
    }
  
    // Getters
    getPhotoshootID() {
      return this.photoshootID;
    }
  
    getDate() {
      return this.date;
    }
  
    // Setters
    setPhotoshootID(photoshootID) {
      this.photoshootID = photoshootID;
    }
  
    setDate(date) {
      this.date = date;
    }
  }
  
  export default Photoshoot;
  